### ISSUE TITLE
Text of upgrade prompt part 2

### ISSUE THREAD

**Issue by obra** [keyboardio/Chrysalis#218]
> There's a possibility that we misdetected the capabilities of the keyboard, or that the keyboard was starting up while we connected. In this case, you can try clicking the "Reconnect" button to attempt a reconnect, and look for the necessary features again.

If this is really true, just automate it. Have chrysalis wait 5 seconds and try a second time. If that doesn't work, the user hitting the button isn't, either.

This also simplifies things for less clueful users.

**Comment by algernon** [comment]
It's not just keyboards booting, but if something else is listening on Serial too, that can interfere with Chrysalis. I usually run a small tool that does that, and end up on this screen, even though my firmware has everything needed - I just have something else stealing traffic. In this case, hitting Reconnect does help. I'm using it all the time, but then I'm an outlier in this respect, and we shouldn't optimize or design for me.

I guess retrying a few times, with a few seconds inbetween doesn't hurt, but we should display something while that's ongoing. Something more than a progress bar.

**Comment by algernon** [comment]
I've been thinking about this a lot in the past few days. I tried to come up with scenarios where we misdetect keyboard features, and found only two:

- When the keyboard is booting up still
- When there's something else listening on the port, and stealing traffic from us

Retries in the second case are useless, because it needs manual action to shut the other program down, and the chances of someone doing that during our retries is slim to none.

Retries in the first case are more useful, but come with a big drawback: connecting to a keyboard that's either booting, or genuinely doesn't support Focus would take considerably more time. The only keyboard where I noticed the slow bootup was the ErgoDox EZ, the rest boot up faster than the time between a human is able to select it from the dropdown and click on connect.

Thus, optimizing for this rare case at the cost of making the more common case (connecting to an Focus-less board) seems not worth it. It'd be a non-negligible effort to implement the code, and the provided user experience would - in my opinion - be worse in the more common case.
